### PR TITLE
Toolchain/nix: Improve the nix development shells

### PR DESCRIPTION
**Pull request by iniw**

### CODE REVIEW

**Comment by nico** [comment]
Is the formatting done with some kind of auto formatter?

**Comment by iniw** [comment]
> Is the formatting done with some kind of auto formatter?
Yes! It was done by [nixfmt](https://github.com/NixOS/nixfmt) under default settings, which is the "official" nix formatter AFAIK.

**Comment by timschumi** [comment]
> > Is the formatting done with some kind of auto formatter?
> 
> Yes! It was done by [nixfmt](https://github.com/NixOS/nixfmt) under default settings, which is the "official" nix formatter AFAIK.
> 

In this case, please codify the formatter in the file. I as far as I know that works fine for devshells.

PS: I started updating the toolchain file some time ago, but ran into build issues (and therefore never uploaded it). I'll try and dig out the branch to see if everything I ran into additionally has been resolved.

**Comment by nico** [comment]
> In this case, please codify the formatter in the file. I as far as I know that works fine for devshells.

Do you think that could happen in a follow-up?

**Comment by iniw** [comment]
> > > Is the formatting done with some kind of auto formatter?
> > 
> > 
> > Yes! It was done by [nixfmt](https://github.com/NixOS/nixfmt) under default settings, which is the "official" nix formatter AFAIK.
> 
> In this case, please codify the formatter in the file. I as far as I know that works fine for devshells.
> 

Done. Not sure how useful being able to do `nix fmt` is considering that command tries doing very dumb things like recursing down gitignored folders ([tho apparently that just got deprecated a couple weeks ago](https://github.com/NixOS/nixfmt/issues/151)) hunting for nix files but perhaps it's useful to communicate what formatter we're expecting.

File: flake.nix [hunk 1]
```diff
@@ -6,10 +6,21 @@
...
+    }:
+    utils.lib.eachDefaultSystem (
+      system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+      in
+      {
+        devShells.default = import ./Toolchain { inherit pkgs; };
+        devShells.ladybird = import ./Ladybird { inherit pkgs; };
+        formatter = pkgs.nixfmt-rfc-style;
```
Comment (nico) [inline flake.nix:23]: Can you put formatting and manual changes to this file in separate commits? Reformat commits should do just the reformatting.

Comment (iniw) [reply flake.nix:23]: 👍🏾

Comment (nico) [reply flake.nix:23]: Thanks! Commit message of the first commit still claims that the first commit does it though.

(Sorry for the many iterations 😞)

Comment (iniw) [reply flake.nix:23]: No problem! Fixed now :)

**Pull request merged**